### PR TITLE
Allow lambda to use ecr

### PR DESCRIPTION
**Pull request by ian-hoyle**
This is just to see if the lambda using the ecr can be created
If it works will delete the old draft-metadata validator and replace with this

### CODE REVIEW

File: root_draft_metadata_ecr.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module "draft_metadata_validator_lambda_ecr" {
+  source          = "./da-terraform-modules/lambda"
+  function_name   = "tdr-draft-metadata-validator-2-${local.environment}"
```
Comment (TomJKing) [inline root_draft_metadata_ecr.tf:3]: Could you rename the function: `...-validator-v2-...` Its the pattern used for other revised Lambdas on TDR eg `tdr-yara-av-v2-intg`

File: root_draft_metadata_ecr.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "TDRDraftMetadataValidatorLambdaPolicyECR${title(local.environment)}" = templatefile("./templates/iam_policy/draft_metadata_validator_lambda.json.tpl", {
+      account_id     = var.tdr_account_number
+      environment    = local.environment
+      parameter_name = local.keycloak_backend_checks_secret_name
+      bucket_name    = local.draft_metadata_s3_bucket_name
+      kms_key_arn    = module.s3_internal_kms_key.kms_key_arn
+    })
+  }
+  plaintext_env_vars = {
+    API_URL            = "graphql"
```
Comment (TomJKing) [inline root_draft_metadata_ecr.tf:20]: In the existing Lambda implementation the `API_URL` is set to the full Consignment API URL value.

In the Terraform it is set as: `"${module.consignment_api.api_url}/graphql"`

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of very minor comments. Thanks.

File: root_draft_metadata_ecr.tf (hunk 1 above)
Comment (ian-hoyle) [reply root_draft_metadata_ecr.tf]: changed

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**